appcheck/appcheck.go

```diff
@@ -166,6 +174,38 @@ func (c *Client) VerifyToken(token string) (*DecodedAppCheckToken, error) {
 	return &appCheckToken, nil
 }
 
+func (c *Client) VerifyTokenWithReplayProtection(token string) (*DecodedAppCheckToken, error) {
+	decodedAppCheckToken, err := c.VerifyToken(token)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to verify token: %v", err)
```

Comment:
Since `err` already contains information on why the token is invalid, there is no need to add a prefix. Please return `err` directly.